hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeHandler.java

```diff
@@ -36,9 +38,12 @@ public interface UpgradeHandler {
    * @param context                instance of {@link HoodieEngineContext} to be used.
    * @param instantTime            current instant time that should not be touched.
    * @param upgradeDowngradeHelper instance of {@link SupportsUpgradeDowngrade} to be used.
-   * @return Map of config properties and its values to be added to table properties.
+   * @return Map of config properties and its values to be added to table properties,
+   *         and a list configs to remove.
    */
-  Map<ConfigProperty, String> upgrade(
-      HoodieWriteConfig config, HoodieEngineContext context, String instantTime,
+  Pair<Map<ConfigProperty, String>, List<ConfigProperty>> upgrade(
```

Comment:
can we create a pojo for this. 

@yihua may be we should land lombok first.. from the other PR..

Reply:
sure. fixed it in latest commit